components/gh_events/GitHubEvents.tsx

```diff
@@ -21,29 +20,24 @@ const excludeBlacklistedEvents = (event: IGitHubEvent) => {
   return !blacklist.includes(event.type);
 };
 
-export default function GitHubEvents({ minimal }: { minimal?: boolean }) {
-  const [page, setPage] = useState(1);
-  const [events, setEvents] = useState<IGitHubEvent[]>();
-
-  useEffect(() => {
-    fetch(
-      `https://api.github.com/orgs/${env.NEXT_PUBLIC_GITHUB_ORG}/events?per_page=100&page=${page}`,
-    )
-      .then((res) => res.json())
-      .then((data) =>
-        setEvents(
-          combineSimilarPushEvents(
-            data.filter(exludeBotEvents).filter(excludeBlacklistedEvents),
-          ).slice(0, 5),
-        ),
-      );
-  }, [page]);
+export default async function GitHubEvents({ minimal }: { minimal?: boolean }) {
+  const events = await getEvent().then((data) =>
+    data
+      .filter(excludeBlacklistedEvents as any)
+      .filter(exludeBotEvents as any)
+      .slice(0, 5),
+  );
+  if (!env.NEXT_PUBLIC_GITHUB_ORG) {
+    throw new Error("Missing NEXT_PUBLIC_GITHUB_ORG");
+  }
```

Comment:
Let's not do this here. Ideally, this should be done at by env schema validation at build time and not during rendering. 

```suggestion
```